File: src/_01_CreationalPattern/_01_03_AbstractFactory/whiteProPartsFactory.js
import ShipPartsFactory from "./shipPartsFactory.js";
import WhiteProAnchor from "./whiteProAnchor.js";
import WhiteProWheel from "./whiteProWheel.js";

/**
 * @typedef { import("./Anchor").default } Anchor
 * @typedef { import("./Wheel").default } Wheel
 */

export default class WhiteProPartsFactory extends ShipPartsFactory {
  constructor() {
    super();
  }

  /** @returns { Anchor } */
  createAnchor() {
    return new WhiteProAnchor("흰색 Pro Anchor");
  }

  /** @returns { Wheel } */
  createWheel() {
    return new WhiteProWheel("White Pro Wheel");
  }
}

File: src/_01_CreationalPattern/_01_03_AbstractFactory/java/after/WhiteProPartsFactory.java
package _01_CreationalPattern._01_03_AbstractFactory.java.after;

public class WhiteProPartsFactory implements ShipPartsFactory {

  @Override
  public Anchor createAnchor() {
    return new WhiteProAnchor();
  }

  @Override
  public Wheel createWheel() {
    return new WhiteProWheel();
  }
  
}
